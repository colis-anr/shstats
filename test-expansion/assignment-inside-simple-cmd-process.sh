#!/bin/sh

x=1
# the effects of the assignment to x and to z are not visible to the shell
# on the same line, Since we have a process creation only the assignment
# to z is effective on the following line.
x=2 y=${z:=3} /bin/echo $x $z
echo $x
echo $z
