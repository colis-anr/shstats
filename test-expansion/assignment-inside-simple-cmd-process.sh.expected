[
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix",
                [
                  "CmdPrefix_AssignmentWord",
                  [ [ "Name", "x" ], [ "Word", "1", [] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix_CmdWord_CmdSuffix",
                [
                  "CmdPrefix_CmdPrefix_AssignmentWord",
                  [
                    "CmdPrefix_AssignmentWord",
                    [ [ "Name", "x" ], [ "Word", "2", [] ] ]
                  ],
                  [
                    [ "Name", "y" ],
                    [ "Word", "${z:=3}", [ [ "WordLiteral", "${z:=3}" ] ] ]
                  ]
                ],
                [ "CmdWord_Word", [ "Word", "/bin/echo", [] ] ],
                [
                  "CmdSuffix_CmdSuffix_Word",
                  [ "CmdSuffix_Word", [ "Word", "1", [] ] ],
                  [ "Word", "$z", [ [ "WordLiteral", "$z" ] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName_CmdSuffix",
                [
                  "CmdName_Word",
                  [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                ],
                [ "CmdSuffix_Word", [ "Word", "1", [] ] ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName_CmdSuffix",
                [
                  "CmdName_Word",
                  [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                ],
                [
                  "CmdSuffix_Word",
                  [ "Word", "$z", [ [ "WordLiteral", "$z" ] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ]
]